pkg/kubectl/cmd/auth/cani.go

```diff
@@ -167,9 +197,28 @@ func (o *CanIOptions) Validate() error {
 			return fmt.Errorf("NonResourceURL and ResourceName can not specified together")
 		}
 	}
+
+	if o.NoHeaders {
```

Comment:
this can't be set w/o list flag, yet you're not checking its value so how do you know it wasn't set?

Reply:
@soltysh I checked List flag in Line 185

Reply:
Right, sorry I missed the `return nil` after the internal condition.